### PR TITLE
Implement more interfaces in dualtor_utils

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <bingwang@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)
This PR add several new interfaces and fixtures in dualtor_utils.
Functions:
```
toggle_simulator_port_to_tor_a
toggle_simulator_port_to_tor_b
check_simulator_read_side
```
Fixtures:
```
toggle_all_simulator_ports_to_tor_a
toggle_all_simulator_ports_to_tor_b
```
The next plan is to toggle y_cable simulator ports with these newly implemented functions and fixtures.
### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR is to add several new interfaces and fixtures in dualtor_utils.

#### How did you verify/test it?
Verified by a demo script, which toggled all ports and verified returned status.

#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
No.

### CODE REVIEW

File: tests/common/dualtor/dualtor_utils.py [hunk 1]
```diff
@@ -31,25 +31,56 @@ def mux_server_url(request, tbinfo):
...
     Helper function to build an url for given port and target
 
     Args:
         server_url: a str, the url for mux server, like http://10.0.0.64:8080/mux/vms17-8
         physical_port: physical port on switch, an integer starting from 1
-        action: a str, either "output" or "drop"
+                        If physical_port is none, the returned url contains no '/port/action' (For polling/toggling all ports)
+        action: a str, output/drop or None. If action is None, the returned url contains no '/action'
     Returns:
         The url for posting flow update request, like http://10.0.0.64:8080/mux/vms17-8/1/drop(output)
```
Comment (wangxin) [inline tests/common/dualtor/dualtor_utils.py:44]: I think the convention is to use "|" , for example "this|that", to indicate either "this" or "that". The example would be better like this:
`http://10.0.0.64:8080/mux/vms17-8/1/drop|output`

File: tests/common/dualtor/dualtor_utils.py [hunk 2]
```diff
@@ -31,25 +31,56 @@ def mux_server_url(request, tbinfo):
...
+        logger.warn("get request returns err. err_msg = {}".format(str(e)))
+    return None
+
 def _post(server_url, data):
     """
     Helper function for posting data to y_cable server.
 
     Args:
-        server_url: a str, the full address of mux server, like http://10.0.0.64:8080/mux/vms17-8/1/drop(output)
+        server_url: a str, the full address of mux server, like http://10.0.0.64:8080/mux/vms17-8/[1/drop(output)]
```
Comment (wangxin) [inline tests/common/dualtor/dualtor_utils.py:83]: I think the convention is to use "|" , for example "this|that", to indicate either "this" or "that".

File: tests/common/dualtor/dualtor_utils.py [hunk 3]
```diff
@@ -112,3 +173,44 @@ def recover_all_directions(mux_server_url, physical_port):
...
+    pytest_assert(_post(server_url, data), "Failed to toggle all ports to TOR A")
+
+@pytest.fixture(scope='module')
+def toggle_all_simulator_ports_to_tor_b(mux_server_url):
+    """
+    A module level fixture to toggle all ports to tor_b
+    """
+    server_url = _url(mux_server_url)
+    data = {"active_side": TOR_B}
+    pytest_assert(_post(server_url, data), "Failed to toggle all ports to TOR A")
```
Comment (wangxin) [inline tests/common/dualtor/dualtor_utils.py:215]: I think this function is dependent on PR https://github.com/Azure/sonic-mgmt/pull/2703.

The simulator server also supports two other options:
* "toggle": Just toggle the active side for all ports
* "random": Randomly set active side for each of the port.

Can you consider adding fixtures for these two options as well?

**Comment by bingwang-ms** [comment]
All updated. Thanks for your great suggestions! @wangxin

File: tests/common/dualtor/dualtor_utils.py [hunk 4]
```diff
@@ -112,3 +173,64 @@ def recover_all_directions(mux_server_url, physical_port):
...
+    then it will be toggled to tor_b.
+    """
+    server_url = _url(mux_server_url)
+    data = {"active_side": "toggle"}
+    pytest_assert(_post(server_url, data), "Failed to toggle all ports to another side")
+
+@pytest.fixture(scope='module')
+def toggle_all_simulator_ports_to_random_side(mux_server_url):
+    """
+    A module level fixture to toggle all ports to an random side.
```
Comment (wangxin) [inline tests/common/dualtor/dualtor_utils.py:231]: -> to a random side

**Review by wangxin — APPROVED** [review]
LGTM except a typo in docstring.

File: tests/common/dualtor/dualtor_utils.py (hunk 4 above)
Comment (bingwang-ms) [reply tests/common/dualtor/dualtor_utils.py]: Thanks. Updated.

**Pull request merged**